### DISCUSSION TITLE
Show process running after a button is clicked

### DISCUSSION THREAD

**Discussion by ReZeroE** [TomSchimansky/CustomTkinter#1774] [General]
I'm looking for a way to display the progress of a function that is connected to a CTkButton. 

For example, if I have the button:

```python
self.button = customtkinter.CTkButton(self, text="Button", command=self.run_process)
```
and the connected function:
```python
def run_process(self):
    time.sleep(10)
    print("done")
```

When the `run_process()` function is being executed, the UI freezes with the button pressed down until `run_process()` finishes executing. Is there a way to "unfreeze" the UI on-screen during the function's execution or show the processing status?

**Comment by Akascape** [comment]
@ReZeroE Use threading at this point.
```python
import threading
...
self.button = customtkinter.CTkButton(self, text="Button", command=lambda: threading.Thread(target=self.run_process).start())
...
```
